### PR TITLE
Fix log4j config for :qa:os tests

### PR DESCRIPTION
**Pull request by pugnascotia**
The log4j config in `:qa:os` was broken because it referenced an appender plugin that is not on that project's classpath. Resolve this by filtering out references to this appender when coping the config from `:test:framework`.

### CODE REVIEW

File: qa/os/build.gradle [hunk 1]
```diff
@@ -62,7 +62,9 @@ tasks.register('destructivePackagingTest') {
 }
 
 processTestResources {
-  from project(":test:framework").file("src/main/resources/log4j2-test.properties")
+  from(project(":test:framework").file("src/main/resources/log4j2-test.properties")) {
```
Comment (mark-vieira) [inline qa/os/build.gradle:65]: Should we even need to share this config? Why not just have a separate config for these tests? Also, so far as I can tell, we don't really use anything from `:test:framework` anyhow. All packaging tests seem to rely only on vanilla junit and randomized runner.

**Review by mark-vieira — CHANGES_REQUESTED** [review]

File: qa/os/build.gradle (hunk 1 above)
Comment (pugnascotia) [reply qa/os/build.gradle]: Aye, I was just being really cautious, but it's probably unnecessary. I've replaced it with a dedicated file.

**Review by mark-vieira — APPROVED** [review]
👍

**Comment by pugnascotia** [comment]
@elasticmachine update branch

**Comment by pugnascotia** [comment]
Backported to `7.x` in 842c9c79d79

**Pull request merged**